### PR TITLE
fix incorrect documentation about service endpoints and add more

### PR DESCRIPTION
**Pull request by albertshau**
details regarding expanded dependencies and automator types.

### CODE REVIEW

File: docs/docs/source/rest/services.rst [hunk 1]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+and choose which automator type will be used to carry out that action. Depending
+on the automator type, different fields may need to be provided. Required fields
+are defined by the automator type, which is given by the plugin itself.
+
+The provisioner section itself is a JSON object with ``actions`` as a key, and a
+JSON object as its value.  The value contains a mapping of service action 
+(one of install, configure, initialize, start, stop, and remove) to 
+automator details. The automator details is a JSON object containing a ``type``,
+which defines which automator type to use for that action, and ``fields``. Fields
+contains another JSON object which is a set of key values pairs required by the
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: "key-value"

File: docs/docs/source/rest/services.rst [hunk 2]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+The provisioner section itself is a JSON object with ``actions`` as a key, and a
+JSON object as its value.  The value contains a mapping of service action 
+(one of install, configure, initialize, start, stop, and remove) to 
+automator details. The automator details is a JSON object containing a ``type``,
+which defines which automator type to use for that action, and ``fields``. Fields
+contains another JSON object which is a set of key values pairs required by the 
+automator plugin.
+
+This example is taken from an example Hadoop namenode service. It uses chef as the 
+automator type, which requires a script field and also allows an optional data field.
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: My pending changes rename the fields from "script, data" to "run_list, json_attributes".

File: docs/docs/source/rest/services.rst [hunk 3]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+                "data": "{\"loom\": { \"node\": { \"services\": { \"hadoop-hdfs-namenode\": \"stop\" } } } }",
+                "script": "recipe[hadoop_wrapper::default],recipe[hadoop::hadoop_hdfs_namenode],recipe[loom_service_runner::default]"
+            }
+        }
+    }
+
+Dependencies
+^^^^^^^^^^^^
+
+Dependencies serve two general purposes. The first is to enforce that a service that requires other services
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: IMO, remove "that requires other services"... redundant and too wordy

File: docs/docs/source/rest/services.rst [hunk 4]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+^^^^^^^^^
+The ``conflicts`` key specifies an array of services that cannot be placed on a cluster with the given service.
+In this example, that means that "myapp-2.0" cannot be placed on a cluster with "myapp-1.0" or "myapp-1.5".
+
+Install
+^^^^^^^
+Install defines install time dependencies. Install time dependencies take effect for the INSTALL and REMOVE
+service actions. It contains a ``requires`` key which specifies an array of services that the given services
+requires for its installation. In this example, "myapp-2.0" requires the "base" service at install time. This
+means that the installation of the "base" service will occur before the install of the "base" service. Similarly,
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: ...will occur before the install of the "myapp-2.0" service

File: docs/docs/source/rest/services.rst [hunk 5]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+the removal of the "myapp-2.0" service will occur before the removal of the "base" service. This also means that
+the "myapp-2.0" service cannot be placed on a service without the "base" service also being present on the cluster.
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: ...placed on a cluster

File: docs/docs/source/rest/services.rst [hunk 6]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+at install time. This means that if the "ntp" service is also on the cluster, the installation of "ntp" will occur
+before the installation of "myapp-2.0". However, "ntp" does not have to be placed on the cluster in order for "myapp-2.0"
+to be placed on the cluster.
+
+Runtime
+^^^^^^^
+Runtime defines run time dependencies. It also contains ``requires`` and ``uses`` keys that are analagous to those
+in the install section. The only difference is the service actions that they apply to. Install dependencies affect
+the INSTALL and REMOVE service actions, whereas runtime dependencies affect the INITIALIZE, START, and STOP dependencies.
+In this example, "myapp-2.0" requires the "sql-db" service. This means that "myapp-2.0" cannot be placed on a cluster
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: we should bold/italicize/whatever the word "requires" and "uses" below, as it is easy to not realize we are referencing the dependency type

File: docs/docs/source/rest/services.rst [hunk 7]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+without a "sql-db" service. It also means that the initialization of "myapp-2.0" will occur after the start of "sql-db".
+It also means the start of "myapp-2.0" will occur after the start of "sql-db". It also means that the stop of "myapp-2.0"
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: combine:  "It also means the start of "myapp-2.0" will occur after the start of "sql-db", and the stop of "myapp-2.0" will occur before the stop of "sql-db"

File: docs/docs/source/rest/services.rst [hunk 8]
```diff
@@ -23,11 +23,143 @@ REST API: Services
...
+^^^^^^^
+Runtime defines run time dependencies. It also contains ``requires`` and ``uses`` keys that are analagous to those
+in the install section. The only difference is the service actions that they apply to. Install dependencies affect
+the INSTALL and REMOVE service actions, whereas runtime dependencies affect the INITIALIZE, START, and STOP dependencies.
+In this example, "myapp-2.0" requires the "sql-db" service. This means that "myapp-2.0" cannot be placed on a cluster
+without a "sql-db" service. It also means that the initialization of "myapp-2.0" will occur after the start of "sql-db".
+It also means the start of "myapp-2.0" will occur after the start of "sql-db". It also means that the stop of "myapp-2.0" 
+will occur before the stop of "sql-db". Similarly, because "myapp-2.0" uses "load-balancer", initialization and start of
+"myapp-2.0" will occur after the start of "load-balancer". Similarly, the stop of "myapp-2.0" will occur before the stop
+of "load-balancer". Since it is in ``uses``, enforcement of this ordering only applies is "load-balancer" is present on the
```
Comment (dereklwood) [inline docs/docs/source/rest/services.rst]: ... applies if...

**Comment by dereklwood** [comment]
Some minor comments... only significant thing is that the "script" and "data" fields for the chef automator plugin are changing to "run_list" and "json_attributes" with the plugin registry stuff.  Since that may be a bigger docs change I can do that in a separate PR also.  Otherwise :+1: lgtm

File: docs/docs/source/rest/services.rst (hunk 1 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: done

File: docs/docs/source/rest/services.rst (hunk 3 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: good catch, i agree. done

File: docs/docs/source/rest/services.rst (hunk 4 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: done

File: docs/docs/source/rest/services.rst (hunk 5 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: done

File: docs/docs/source/rest/services.rst (hunk 6 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: done

File: docs/docs/source/rest/services.rst (hunk 7 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: done

File: docs/docs/source/rest/services.rst (hunk 8 above)
Comment (albertshau) [inline docs/docs/source/rest/services.rst]: done

**Comment by albertshau** [comment]
changed the fields to run_list and json_attributes as well.
